repo/tuf_on_ci/_repository.py

```diff
@@ -281,9 +281,34 @@ def _validate_role(
         if md.signed.expires > datetime.utcnow() + timedelta(days=days):
             return False, f"Expiry date is further than expected {days} days ahead"
 
+        if isinstance(md.signed, Root):
+            # tuf-on-ci is always consistent_snapshot
+            if not md.signed.consistent_snapshot:
+                return False, "Consistent snapshot is not enabled"
+
+            # Specification: root version must be x+1, not just larger
+            if prev_md and prev_md.signed != md.signed:
```

Comment:
Haven't looked at how you're getting `prev_md`, but isn't there a consistency issue if `prev_md.signed == md.signed`?

Reply:
I think there are cases where we check the validity of roles even if the metadata has not (yet) changed in this PR. In that case we wouldn't want to enforce the version bump requirement.